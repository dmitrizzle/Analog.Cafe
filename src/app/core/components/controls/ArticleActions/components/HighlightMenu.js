import { ButtonStrip } from "@roast-cms/react-button-beans"
import React from "react"
import styled from "styled-components"
import open from "oauth-open"

import {
  brandButtonStyles,
  twitterButtonStyles
} from "../../Button/components/SocialButtons"
import ButtonStripItem from "../../Button/components/ButtonStripItem"
import Twitter from "../../../icons/Social/components/Twitter"

export const Menu = styled(ButtonStrip)`
  display: ${props => (props.hidden ? "none" : "block")};
  position: absolute;
  z-index: ${props => props.theme.layer.up + 1};
`

const MenuItem = styled(ButtonStripItem)`
  ${brandButtonStyles}${twitterButtonStyles};
  background: ${props => props.theme.color.foreground()};
  width: 2em;
  padding-left: 0;

  svg {
    margin-left: 0
  }
  ${props => props.theme.size.breakpoint.max.s`
    width: 2.5em;
    height: 2em;
    border-radius: ${props => props.theme.effects.borderRadius.med}em;
    svg {
      height: 2.5em;
      margin-top: -0.9em;
    }
  `};
`

export default props => (
  <Menu
    key="App_SelectionNav"
    hidden={props.selection.hidden}
    style={{
      top: `${props.selection.topOffset - 30}px`,
      left: `${props.selection.leftOffset - 20}px`
    }}
  >
    <div>
      <MenuItem
        left
        right
        onClick={event => {
          event.preventDefault()
          event.stopPropagation()
          open(
            `https://twitter.com/intent/tweet?text=${props.selection.text}`,
            () => {}
          )
        }}
      >
        <Twitter />
      </MenuItem>
    </div>
  </Menu>
)
