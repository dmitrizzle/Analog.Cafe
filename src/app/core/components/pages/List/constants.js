export const VALUE_ZIGZAG_WIDTH = "33%"
export const VALUE_ZIGZAG_TOP_SHIM = 12
