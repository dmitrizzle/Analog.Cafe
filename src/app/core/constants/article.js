// related constants
import { ROUTE_API_DOMAIN } from "./app"

// api endpoint
export const ROUTE_ARTICLE_API = ROUTE_API_DOMAIN + "/articles"
export const ROUTE_SUBMISSION_API = ROUTE_API_DOMAIN + "/submissions"

// user-facing URL
export const ROUTE_ARTICLE_DIR = "/zine"
export const ROUTE_SUBMISSIONS_DIR = "/submissions"
