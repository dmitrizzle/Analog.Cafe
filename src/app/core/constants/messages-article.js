export const DOCUMENT_PLACEHOLDER = {
  raw: {
    document: {
      nodes: [
        {
          object: "block",
          type: "paragraph",
          nodes: [
            {
              object: "text",
              leaves: [
                {
                  text:
                    "█████████ █████ █████████ ██████████████ ███████████ ████████ █████ ██ █████████ █████ █████████ ██████████████ █████████████ ██████ ███ ██ ██████ █████ █████ █████████ ██████████████ ███████████ ████████ █████ ███████████ █████ █████████ █████████ █████ ███████████ ████████ █████ ███████████ █████ █████████ ██████████████ ███████████ ████████ █████ ███████████ █████ █████████ ██████████ █████████ █████ █████████ █ ████████ ████████████ █████████ ███████████ █████████████ █ ████████ █████ ██"
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
