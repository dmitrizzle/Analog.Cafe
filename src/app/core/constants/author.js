// related constants

// api endpoint
import { ROUTE_API_DOMAIN } from "../../constants"

export const ROUTE_AUTHOR_API = ROUTE_API_DOMAIN + "/authors"
