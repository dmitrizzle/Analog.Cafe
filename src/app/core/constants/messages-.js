import { TEXT_EMOJIS } from "../../constants"

// not all objects below belong in "card" category, need to be refactored
// same for a few other constants files - do have a look.
// some of this may need to move to the root folder as it's being used by all of
// the app.
export const CARD_ERRORS = {
  ARTICLE: {
    title: TEXT_EMOJIS.WTF,
    subtitle: "Page Not Available"
  },
  LIST: {
    title: "Nothing here yet",
    emoji: TEXT_EMOJIS.WTF
  },
  LIST_OFFLINE: {
    title: "You aren’t connected to the internet",
    emoji: TEXT_EMOJIS.WTF
  },
  PICTURE_AUTHOR: {
    name: "Unknown Author",
    title: "Info not available",
    text: "This image has been authored by someone not listed in our records…"
  },
  CARD: {
    title: "Info Not Available",
    text: "This card could not be loaded…"
  },
  AUTHOR: {
    text: "Author hasn’t shared any details about her- or himself yet."
  }
}
export const TEXT_ERRORS = {
  CODE_103: { error: "Error: User already authenticated. (103)" },
  CODE_204: { error: "Error: Malformed or no data received. (204)" },
  CODE_404: {
    error: "Error: This view or data for this view does not exist. (404)"
  },
  CODE_403: {
    error: "Error: Viewing this content requires you to sign in. (403)"
  },
  CODE_401: {
    error: "Error: You need to sign in to access your account. (401)",
    TokenExpiredError:
      "You have been automatically signed out, please sign in again. (401)",
    JsonWebTokenError:
      "You will need to sign in again should you want to submit or edit your account. (401) "
  }
}
export const TEXT_LABELS = {
  SUBSRIBE: "Subscribe ❤︎",
  SUBMIT: "Submit ❤︎"
}
