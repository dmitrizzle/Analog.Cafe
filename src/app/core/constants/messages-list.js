import { APP_DESCRIPTION } from "../../constants"
import { ROUTE_URL_USER_LANDING } from "../../user/constants/routes-session"

export const TEXT_STATUS_LABELS = {
  pending: "In Queue",
  rejected: "Not Published",
  unpublished: "Unpublished",
  scheduled: "Scheduled",
  published: "Published"
}
export const TEXT_ROUTE_LABELS = {
  [ROUTE_URL_USER_LANDING]: {
    title: "Your submissions to Analog.Cafe"
  },
  "/stories": {
    title: "Stories: any medium, any topic",
    description:
      "Stories, essays and creative writing, beyond film photography."
  },
  "/editorials": {
    title: "Notes from Analog.Cafe editors",
    description:
      "Announcements, notes and opinion pieces from Analog.Cafe editors."
  },
  "/guides": {
    title: "Guides: learn, understand & DIY",
    description:
      "Guides, explainations and instructables on film photography, art and technology."
  },
  "/reviews": {
    title: "Reviews: tools, toys & places",
    description:
      "Film photography gear, creative tools, places & experiences that inspire."
  },
  "/photo-essays": {
    title: "Photo essays; shot on film",
    description:
      "A colleciton of photo essays created with analog (film) cameras."
  },
  "/articles": {
    title: "Stories, reviews, guides & editorials",
    description:
      "A collection of articles about art, inspiration and technology."
  },
  "/collaborations": {
    title: "Collaborations",
    description:
      "Collaborations are written pieces with images created by multiple artists and photographers."
  },
  "/solo-projects": {
    title: "Solo projects",
    description:
      "Solo projects are written pieces with images created by a single author."
  },
  "/": {
    title: APP_DESCRIPTION,
    description:
      "Analog.Cafe is an inclusive creative outlet that publishes outstanding images and stories.",
    designation: "homepage"
  },
  "/author/*": {
    title: "Images & stories",
    description: "Published works and photography by Analog.Cafe author."
  },
  default: {
    title: APP_DESCRIPTION
  }
}

export const LIST_PLACEHOLDER = [
  {
    type: "placeholder",
    tag: "████████",
    title: "███",
    id: "0000000",
    author: {
      name: "██████"
    },
    summary: "█ ████████ ██████ ████ ████ ██████████ ████ ████████████████ ██"
  },
  {
    type: "placeholder",
    tag: "████",
    title: "████",
    id: "0000001",
    author: {
      name: "█████"
    },
    summary: "█████ ████████████ ████████ ███ ███████████ ██████████████████"
  },
  {
    type: "placeholder",
    tag: "█████████",
    title: "██",
    id: "0000002",
    author: {
      name: "██"
    },
    summary: "█ ██████████ ██████ ██████ ███████████ ███ ███████████ █████████"
  }
]
