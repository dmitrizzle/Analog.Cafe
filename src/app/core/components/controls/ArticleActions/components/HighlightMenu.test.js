import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import { APP_THEME } from "../../../../../../constants"
import HighlightMenu, { MenuItem, Menu } from "./HighlightMenu"

it("Render HighlightMenu without crashing, matches snapshot", () => {
  const element = shallow(
    <HighlightMenu selection={{ hidden: false, topOffset: 0, leftOffset: 0 }} />
  )
  expect(element).toMatchSnapshot()
})
it("Render MenuItem without crashing, matches snapshot", () => {
  const element = shallow(<MenuItem theme={APP_THEME} />)
  expect(element).toMatchSnapshot()
})
it("Render Menu without crashing, matches snapshot", () => {
  const element = shallow(<Menu theme={APP_THEME} />)
  expect(element).toMatchSnapshot()
})
