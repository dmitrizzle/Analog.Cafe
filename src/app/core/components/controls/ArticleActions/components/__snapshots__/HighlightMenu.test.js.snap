// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HighlightMenu without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selection={
        Object {
            "hidden": false,
            "leftOffset": 0,
            "topOffset": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "App_SelectionNav",
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Styled(Component)
                left={true}
                onClick={[Function]}
                right={true}
        >
                <Unknown />
        </Styled(Component)>
</div>,
      "hidden": false,
      "style": Object {
        "left": "-20px",
        "top": "-30px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Styled(Component)
          left={true}
          onClick={[Function]}
          right={true}
>
          <Unknown />
</Styled(Component)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown />,
          "left": true,
          "onClick": [Function],
          "right": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "App_SelectionNav",
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Styled(Component)
                    left={true}
                    onClick={[Function]}
                    right={true}
          >
                    <Unknown />
          </Styled(Component)>
</div>,
        "hidden": false,
        "style": Object {
          "left": "-20px",
          "top": "-30px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Styled(Component)
            left={true}
            onClick={[Function]}
            right={true}
>
            <Unknown />
</Styled(Component)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown />,
            "left": true,
            "onClick": [Function],
            "right": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render Menu without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(styled.div)
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "sc-gzVnrw kFgdUJ",
      "innerRef": undefined,
      "theme": Object {
        "color": Object {
          "background": [Function],
          "brand": [Function],
          "foreground": [Function],
          "highlight": [Function],
        },
        "effects": Object {
          "borderRadius": Object {
            "med": 0.5,
            "small": [Function],
          },
        },
        "elements": Object {
          "thickBorder": [Function],
        },
        "layer": Object {
          "card": 30,
          "nav": 20,
          "overlay": 40,
          "tuck": -1,
          "up": 10,
        },
        "opacity": Object {
          "half": 0.5,
          "least": 0.125,
          "most": 0.85,
        },
        "size": Object {
          "block": Object {
            "border": 8,
            "column": Object {
              "l": 820,
              "m": 750,
            },
            "padding": 1.5,
            "spacing": 1,
          },
          "breakpoint": Object {
            "exact": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "max": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "min": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "stops": Object {
              "max": 1600,
              "min": 320,
            },
          },
          "font": Object {
            "auto": Array [
              "
        ",
              "
		@media (min-width: ",
              "32.5625",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "32.5",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "20",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (min-width: ",
              "100.0625",
              "em) {
			",
              "font-size:	",
              [Function],
              "px;",
              "
		}
	",
              ";
      ",
            ],
            "l": 23,
            "m": 20,
            "make": Object {
              "larger": 3,
              "normal": 1,
              "smaller": 0.85,
              "tiny": 0.65,
            },
            "s": 18,
            "xs": 17,
          },
        },
        "typography": Object {
          "text": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "'Lora', Georgia, serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
        ",
            ],
            "font": "'Lora', Georgia, serif",
            "lineHeight": 1.75,
          },
          "title": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "'Exo 2', Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
          font-weight: ",
              "600",
              ";
          margin: 0;
        ",
            ],
            "font": "'Exo 2', Arial, sans-serif",
            "lineHeight": 1.15,
          },
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "sc-gzVnrw kFgdUJ",
        "innerRef": undefined,
        "theme": Object {
          "color": Object {
            "background": [Function],
            "brand": [Function],
            "foreground": [Function],
            "highlight": [Function],
          },
          "effects": Object {
            "borderRadius": Object {
              "med": 0.5,
              "small": [Function],
            },
          },
          "elements": Object {
            "thickBorder": [Function],
          },
          "layer": Object {
            "card": 30,
            "nav": 20,
            "overlay": 40,
            "tuck": -1,
            "up": 10,
          },
          "opacity": Object {
            "half": 0.5,
            "least": 0.125,
            "most": 0.85,
          },
          "size": Object {
            "block": Object {
              "border": 8,
              "column": Object {
                "l": 820,
                "m": 750,
              },
              "padding": 1.5,
              "spacing": 1,
            },
            "breakpoint": Object {
              "exact": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "max": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "min": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "stops": Object {
                "max": 1600,
                "min": 320,
              },
            },
            "font": Object {
              "auto": Array [
                "
        ",
                "
		@media (min-width: ",
                "32.5625",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (max-width: ",
                "32.5",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (max-width: ",
                "20",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (min-width: ",
                "100.0625",
                "em) {
			",
                "font-size:	",
                [Function],
                "px;",
                "
		}
	",
                ";
      ",
              ],
              "l": 23,
              "m": 20,
              "make": Object {
                "larger": 3,
                "normal": 1,
                "smaller": 0.85,
                "tiny": 0.65,
              },
              "s": 18,
              "xs": 17,
            },
          },
          "typography": Object {
            "text": Object {
              "auto": Array [
                "
          color: ",
                [Function],
                ";
          font-family: ",
                "'Lora', Georgia, serif",
                ";
          letter-spacing: 0.025em;
          line-height: ",
                [Function],
                "em;
        ",
              ],
              "font": "'Lora', Georgia, serif",
              "lineHeight": 1.75,
            },
            "title": Object {
              "auto": Array [
                "
          color: ",
                [Function],
                ";
          font-family: ",
                "'Exo 2', Arial, sans-serif",
                ";
          letter-spacing: 0.025em;
          line-height: ",
                [Function],
                "em;
          font-weight: ",
                "600",
                ";
          margin: 0;
        ",
              ],
              "font": "'Exo 2', Arial, sans-serif",
              "lineHeight": 1.15,
            },
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render MenuItem without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Component)
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "className": "sc-htoDjs hHDWaS",
      "theme": Object {
        "color": Object {
          "background": [Function],
          "brand": [Function],
          "foreground": [Function],
          "highlight": [Function],
        },
        "effects": Object {
          "borderRadius": Object {
            "med": 0.5,
            "small": [Function],
          },
        },
        "elements": Object {
          "thickBorder": [Function],
        },
        "layer": Object {
          "card": 30,
          "nav": 20,
          "overlay": 40,
          "tuck": -1,
          "up": 10,
        },
        "opacity": Object {
          "half": 0.5,
          "least": 0.125,
          "most": 0.85,
        },
        "size": Object {
          "block": Object {
            "border": 8,
            "column": Object {
              "l": 820,
              "m": 750,
            },
            "padding": 1.5,
            "spacing": 1,
          },
          "breakpoint": Object {
            "exact": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "max": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "min": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "stops": Object {
              "max": 1600,
              "min": 320,
            },
          },
          "font": Object {
            "auto": Array [
              "
        ",
              "
		@media (min-width: ",
              "32.5625",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "32.5",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "20",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (min-width: ",
              "100.0625",
              "em) {
			",
              "font-size:	",
              [Function],
              "px;",
              "
		}
	",
              ";
      ",
            ],
            "l": 23,
            "m": 20,
            "make": Object {
              "larger": 3,
              "normal": 1,
              "smaller": 0.85,
              "tiny": 0.65,
            },
            "s": 18,
            "xs": 17,
          },
        },
        "typography": Object {
          "text": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "'Lora', Georgia, serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
        ",
            ],
            "font": "'Lora', Georgia, serif",
            "lineHeight": 1.75,
          },
          "title": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "'Exo 2', Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
          font-weight: ",
              "600",
              ";
          margin: 0;
        ",
            ],
            "font": "'Exo 2', Arial, sans-serif",
            "lineHeight": 1.15,
          },
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "sc-htoDjs hHDWaS",
        "theme": Object {
          "color": Object {
            "background": [Function],
            "brand": [Function],
            "foreground": [Function],
            "highlight": [Function],
          },
          "effects": Object {
            "borderRadius": Object {
              "med": 0.5,
              "small": [Function],
            },
          },
          "elements": Object {
            "thickBorder": [Function],
          },
          "layer": Object {
            "card": 30,
            "nav": 20,
            "overlay": 40,
            "tuck": -1,
            "up": 10,
          },
          "opacity": Object {
            "half": 0.5,
            "least": 0.125,
            "most": 0.85,
          },
          "size": Object {
            "block": Object {
              "border": 8,
              "column": Object {
                "l": 820,
                "m": 750,
              },
              "padding": 1.5,
              "spacing": 1,
            },
            "breakpoint": Object {
              "exact": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "max": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "min": Object {
                "l": [Function],
                "m": [Function],
                "s": [Function],
                "xl": [Function],
                "xs": [Function],
                "xxl": [Function],
              },
              "stops": Object {
                "max": 1600,
                "min": 320,
              },
            },
            "font": Object {
              "auto": Array [
                "
        ",
                "
		@media (min-width: ",
                "32.5625",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (max-width: ",
                "32.5",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (max-width: ",
                "20",
                "em) {
			",
                "font-size: 	",
                [Function],
                "px;",
                "
		}
	",
                " ",
                "
		@media (min-width: ",
                "100.0625",
                "em) {
			",
                "font-size:	",
                [Function],
                "px;",
                "
		}
	",
                ";
      ",
              ],
              "l": 23,
              "m": 20,
              "make": Object {
                "larger": 3,
                "normal": 1,
                "smaller": 0.85,
                "tiny": 0.65,
              },
              "s": 18,
              "xs": 17,
            },
          },
          "typography": Object {
            "text": Object {
              "auto": Array [
                "
          color: ",
                [Function],
                ";
          font-family: ",
                "'Lora', Georgia, serif",
                ";
          letter-spacing: 0.025em;
          line-height: ",
                [Function],
                "em;
        ",
              ],
              "font": "'Lora', Georgia, serif",
              "lineHeight": 1.75,
            },
            "title": Object {
              "auto": Array [
                "
          color: ",
                [Function],
                ";
          font-family: ",
                "'Exo 2', Arial, sans-serif",
                ";
          letter-spacing: 0.025em;
          line-height: ",
                [Function],
                "em;
          font-weight: ",
                "600",
                ";
          margin: 0;
        ",
              ],
              "font": "'Exo 2', Arial, sans-serif",
              "lineHeight": 1.15,
            },
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
